share/build/buildlib.ekat

```diff
@@ -0,0 +1,156 @@
+
+sys.path.append(os.path.dirname(__file__))
+from buildlib_util import extract_from_macros
+
+logger = logging.getLogger(__name__)
+
+###############################################################################
+def parse_command_line(args, description):
+###############################################################################
+    parser = argparse.ArgumentParser(
+        usage="""\n{0} [--debug]
+OR
+{0} --verbose
+OR
+{0} --help
+
+\033[1mEXAMPLES:\033[0m
+    \033[1;32m# Run \033[0m
+    > {0}
+""".format(
+            os.path.basename(args[0])
+        ),
+        description=description,
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+
+    utils.setup_standard_logging_options(parser)
+
+    parser.add_argument("buildroot", help="build path root")
+
+    parser.add_argument("installpath", help="install path ")
+
+    parser.add_argument(
+        "caseroot", nargs="?", default=os.getcwd(), help="Case directory to build"
+    )
+
+    args = utils.parse_args_and_handle_standard_logging_options(args, parser)
+
+    return args.buildroot, args.installpath, args.caseroot
+
+###############################################################################
+def buildlib(bldroot, installpath, case):
+###############################################################################
+
+    # Figure out where we are getting Kokkos from
+    kokkos_dir = None
+    if case.get_value("USE_ALBANY") or case.get_value("USE_TRILINOS"):
+        print ("case is using Trilinos and will use its Kokkos")
+
+        # Set Kokkos_ROOT so Ekat knows to find Kokkos from Albany/Trilinos
+        expect(os.environ.get("Trilinos_ROOT") is not None,
+               "USE_TRILINOS|USE_ALBANY is On, but Trilinos_ROOT is not set!")
+        os.environ["Kokkos_ROOT"] = os.environ.get("Trilinos_ROOT")
+
+    elif os.environ.get("Kokkos_ROOT") is not None:
+        kokkos_dir = os.environ.get("Kokkos_ROOT")
+
+    installed_ekat_dir = os.environ.get("Ekat_ROOT")
+    if installed_ekat_dir is not None:
+        # We are trying to use a pre-installed ekat. Look for the relevant folders/libs,
+        # and if all looks good, return. Otherwise, crap out
+        ekat_root = os.path.abspath(installed_ekat_dir)
+        include_dir = os.path.join(ekat_root,'include')
+
+        expect (os.path.isdir(ekat_root),f"Non-existent ekat install dir '{ekat_root}'")
+        expect (os.path.isdir(include_dir),f"Missing include subfolder in ekat install dir '{ekat_root}'")
+        expect (os.path.isdir(os.path.join(include_dir,"ekat")),f"Missing ekat headers in '{include_dir}'")
+
+        print (f"Using pre-installed Ekat_ROOT: {ekat_root}")
+        return
+    else:
+        print ("no value found in env for Ekat_ROOT. building from scratch")
+
+    srcroot = case.get_value("SRCROOT")
+    ekat_dir = os.path.join(srcroot, "externals", "ekat")
+    expect(os.path.isdir(ekat_dir), "Missing ekat submodule")
+
+    if kokkos_dir is None:
```

Comment:
I think we would already get an error when cmake runs, as the ekat::KokkosUtils target won't even configure. If you want to check for the kokkos submod here, that's fine. But then: why not check also the presence of spdlog and yaml-cpp (catch2 is prob unused in CIME builds)?